mod-source-record-storage-server/src/main/resources/liquibase/tenant/scripts/v-5.10.0/2024-12-10--13-00-create-and-fill-old-records-tracking-table.xml

```diff
@@ -0,0 +1,62 @@
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.0.xsd">
+
+  <changeSet id="2024-12-10--13-00-create-old-records-tracking-table" author="Volodymyr_Rohach">
+    <createTable tableName="old_records_tracking">
+      <column name="marc_id" type="uuid">
+        <constraints primaryKey="true" nullable="false"/>
+      </column>
+      <column name="has_been_processed" type="boolean">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+  </changeSet>
+
+
+  <changeSet id="2024-12-10--14-00-migrate-old-records" author="Volodymyr_Rohach">
+    <sql>
+      INSERT INTO ${database.defaultSchemaName}.old_records_tracking (marc_id, has_been_processed)
+      SELECT id, FALSE
+      FROM ${database.defaultSchemaName}.records_lb
+      WHERE state = 'OLD'
+        ON CONFLICT (marc_id) DO NOTHING;
+    </sql>
+  </changeSet>
+
+  <changeSet id="2024-12-10--13-00-create-process-old-record-state-change" author="Volodymyr_Rohach">
+    <sql>
+      CREATE OR REPLACE FUNCTION ${database.defaultSchemaName}.track_old_records()
+      RETURNS TRIGGER AS $$
+      BEGIN
+              IF NEW.state = 'OLD' AND (TG_OP = 'INSERT' OR OLD.state IS DISTINCT FROM 'OLD') THEN
```

Comment:
I wonder if old records are ever touched again such that we could simplify the criteria to `NEW.state = 'OLD'`

Reply:
Fixed, thank you!